### PR TITLE
feat: Added type hints to method parameters

### PR DESCRIPTION
**Pull request by lukaszsocha2**
Closes: SDK-1875

### CODE REVIEW

File: boxsdk/auth/oauth2.py [hunk 1]
```diff
@@ -45,53 +49,35 @@ class OAuth2:
...
-            session=None,
-            refresh_lock=None,
+            client_id: Optional[str],
+            client_secret: Optional[str],
+            store_tokens: Callable = None,
+            box_device_id: str = '0',
+            box_device_name: str = '',
+            access_token: str = None,
+            refresh_token: Optional[str] = None,
+            session: Session = None,
```
Comment (arjankowski) [inline boxsdk/auth/oauth2.py:59]: `session: Session = None` -> `session: Optional[Session] = None` ?

File: boxsdk/object/group.py [hunk 2]
```diff
@@ -49,27 +56,24 @@ def get_memberships(self, limit=None, offset=None, fields=None):
         )
 
     @api_call
-    def add_member(self, user, role=GroupRole.MEMBER, configurable_permissions=SDK_VALUE_NOT_SET):
+    def add_member(
+            self,
+            user: 'User',
+            role: str = GroupRole.MEMBER,
```
[the remark this answers is not in the document]
Comment (lukaszsocha2) [reply boxsdk/object/group.py:62]: Yes it should be

File: boxsdk/auth/jwt_auth.py [hunk 3]
```diff
@@ -25,20 +29,20 @@ class JWTAuth(OAuth2):
...
-            rsa_private_key_data=None,
+            client_id: str,
+            client_secret: str,
+            enterprise_id: Optional[str],
+            jwt_key_id: str,
+            rsa_private_key_file_sys_path: str = None,
+            rsa_private_key_passphrase: Optional[bytes] = None,
+            user: Optional[Union[str, 'User']] = None,
+            store_tokens: Callable = None,
+            box_device_id: str = '0',
```
[the remark this answers is not in the document]
Comment (lukaszsocha2) [reply boxsdk/auth/jwt_auth.py:40]: added

File: boxsdk/object/collaboration.py [hunk 4]
```diff
@@ -31,22 +31,20 @@ class Collaboration(BaseObject):
     _item_type = 'collaboration'
 
     @api_call
-    def update_info(self, role=None, status=None):
+    def update_info(
+            self,
+            role: CollaborationRole = None,
```
[the remark this answers is not in the document]
Comment (lukaszsocha2) [reply boxsdk/object/collaboration.py:36]: done

File: boxsdk/object/events.py [hunk 5]
```diff
@@ -93,38 +95,30 @@ def get_events(self, limit=100, stream_position=0, stream_type=UserEventsStreamT
         return self.translator.translate(self._session, response_object=response)
 
     @api_call
-    def get_admin_events(self, limit=None, stream_position=0, created_after=None, created_before=None, event_types=None):
+    def get_admin_events(
+            self,
+            limit: Optional[int] = None,
+            stream_position: Union[str, int] = 0,
+            created_after: str = None,
```
[the remark this answers is not in the document]
Comment (lukaszsocha2) [reply boxsdk/object/events.py:102]: done

File: boxsdk/object/file.py [hunk 6]
```diff
@@ -1,37 +1,41 @@
...
+
 
 class File(Item):
     """Box API endpoint for interacting with files."""
 
     _item_type = 'file'
 
     @api_call
-    def preflight_check(self, size, name=None):
+    def preflight_check(self, size: int, name: str = None) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (lukaszsocha2) [reply boxsdk/object/file.py:28]: done

File: boxsdk/auth/oauth2.py (hunk 1 above)
Comment (lukaszsocha2) [reply boxsdk/auth/oauth2.py:59]: done

**Review by arjankowski — APPROVED** [review]

**Pull request merged**